pkg/util/logger/logger.go

```diff
@@ -15,44 +15,95 @@ type Logger struct {
 }
 
 // Prm groups Logger's parameters.
+// Successful passing non-nil parameters to the NewLogger (if returned
+// errors is nil) connects the parameters with the returned Logger.
+// Parameters that have been connected to the Logger supports configuration
+// changes of the connected Logger.
+//
+// Passing Prm after a successful connection via the NewLogger, connects
+// the Prm to a new instance
+// of the Logger.
+//
+// See also Reload, SetLevelString.
 type Prm struct {
+	// link to the created Logger
+	// instance. Used for a runtime
+	// reconfiguration.
+	_log *Logger
+
+	// support runtime rereading
 	level zapcore.Level
-}
 
-// ErrNilLogger is returned by functions that
-// expect a non-nil Logger but received nil.
-var ErrNilLogger = errors.New("logger is nil")
+	// do not support runtime rereading
+}
 
-// SetLevelString sets the minimum logging level.
+// SetLevelString sets the minimum logging level. Default is
+// "info".
 //
-// Returns error of s is not a string representation of zap.Level
-// value (see zapcore.Level docs).
+// Returns error of s is not a string representation of a
+// supporting logging level.
+//
+// Supports runtime rereading.
 func (p *Prm) SetLevelString(s string) error {
 	return p.level.UnmarshalText([]byte(s))
 }
 
-// NewLogger constructs a new zap logger instance.
+// ErrLoggerNotConnected is returned when a Prm has not been connected to any
+// Logger but Prm.Reload had been called.
+var ErrLoggerNotConnected = errors.New("parameters are not connected to any Logger")
+
+// Reload reloads configuration of a connected instance of the Logger.
+// Returns ErrLoggerNotConnected if no connection has been performed.
+// Returns any reconfiguration errors from the Logger directly.
+func (p Prm) Reload() error {
+	if p._log == nil {
+		return ErrLoggerNotConnected
```

Comment:
When this can happen? Shouldn't we create a new logger in this case?

Reply:
> When this can happen?

```go
var Prm logger.Prm
prm.Reload() // here
```

we can panic since it is totally incorrect usage or return the static err

> Shouldn't we create a new logger in this case?

do you mean creating a new logger inside the `Reload` method and placing a new instance using `p._log` OR it is the general question of what a caller should do?

Reply:
I mean do we have such code anywhere? If this error is not reachable we can panic here, may be even implicitly.

Reply:
> do we have such code anywhere

not yet (there are talks about moving the logger to SDK or a separate repo and sharing it b/w our repos)

> we can panic here

added a panic, but explicit, with the message